packages/backend/index.js

```diff
@@ -78,8 +161,46 @@ app.get("/reviews/", async (req, res) => {
   }
 });
 
+// creating a review
+app.post("/review/:id", async (req, res) => {
+  try {
+    const reviewToAdd = req.body;
+    console.log(reviewToAdd);
+    // get the user of the person trying to write review
+    // note: the username is gotten from the frontend which isn't secure, to update once we get auth running
+    const user = await userServices.findUserByName(
+      reviewToAdd.username
+    );
+    // get the album the user wants to write a review on
+    const album = await userServices.findAlbumById(
+      req.params.id
+    );
+
+    // make a new review with the review schema
+    const newReview = new Reviews({
+      written_by: user._id,
+      rating: reviewToAdd.rating,
+      content: reviewToAdd.content,
+      likes: 0,
+      album_id: album._id
+    });
+    let reviewId;
+
+    await newReview.save().then((review) => {
+      reviewId = review._id;
+    });
+
+    res.status(201);
+    res.send(newReview);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Internal server error" });
+  }
+});
+
+/* Albums */
 // Getting a single album
-app.get("/albums/:id", async (req, res) => {
+app.get("/albums/:id", authenticateUser, async (req, res) => {
```

Comment:
is there a reason why we only do authenticateUser for the albums route, but not for reviews?  would this mean that users that are not authenticated could potentially create reviews without properly signing in for now?

Reply:
I think it's because you can only access the Albums page if you're authenticated, so you can't even access reviews if you're not signed in. Not really sure, I'd like some clarification as well.

Reply:
Good catch! This was leftover from when I was testing the authenticateUser before I had the app run it for all of them. I will make sure to fix that